src/teacher/views.py

```diff
@@ -124,5 +128,9 @@ class TeacherDeleteView(CustomLoginRequiredMixin, DeleteView):
     model = Teacher
     success_url = reverse_lazy('teacher:list')
 
+    def post(self, *args, **kwargs):
+        messages.success(self.request, f'Teacher deleted!')
+        return super().post(*args, **kwargs)
```

Comment:
I would have swapped this 2 lines, just in case something goes wrong and exception is thrown while deleting. In this case you will still display successful message `'Teached deleted!'`.

Reply:
Fixed!